homeassistant/components/esphome/sensor.py

```diff
@@ -74,6 +78,50 @@ async def async_setup_entry(
 class EsphomeSensor(EsphomeEntity[SensorInfo, SensorState], SensorEntity):
     """A sensor implementation for esphome."""
 
+    async def async_added_to_hass(self) -> None:
+        """Register callbacks."""
+        await super().async_added_to_hass()
+
+        if self._static_info.last_reset_type is LastResetType.NEVER:
+            self._attr_last_reset = datetime.min
+        elif self._static_info.last_reset_type is LastResetType.AUTO:
+
+            @callback
+            def check_last_state(event: Event) -> None:
```

Comment:
It would be slightly more efficient to remove the indent and make this `def _check_last_state(self, event: Event) -> None:` since it wouldn't have to recreate function for every entity

Reply:
Done